### DISCUSSION TITLE
Error preparing artifacts: content digest mismatch

### DISCUSSION THREAD

**Discussion by samwise-commits** [getporter/porter#1518] [Q&A]
Hey everyone.

I'm trying to do a 'porter archive' and I'm running into a digest mismatch error. I just pushed that bundle up to our repo last night, yet when I try to archive it today it gives me the digest error. I ran into it yesterday as well, but pushed a new bundle and it didn't throw that error. Though I couldn't even get the archive to work for some reason. The tgz file is blank. Any ideas why I'm getting the digest error or why the archive tgz is blank?

Thanks!

**Comment by carolynvs** [comment]
This is definitely a bug. I have a related one with the same error message for a different command https://github.com/getporter/porter/issues/1445. I've linked our discussion here to that issue. If you have reproduction steps, please let me know. I'll see if I can figure out how to reproduce that content digest mismatch error as well.

I am unsure if the empty tgz file is a symptom of the content digest error. If you have more info on that, I can look into it as well but for now I'm guessing that fixing one will fix the other.

**Reply by samwise-commits** [reply]
This is on a private repo, so unfortunately I can't give the exact repo/bundle I am trying to archive.
But I'm essentially just doing:
`porter archive --reference <reference> file.tgz`

I am also running the latest version (0.35.0) and I'm on Mac - if it makes a difference.

I'm not 100% sure its not something with our repo/bundle, so trying to figure that out as well.
We recently took out the digest tag in our porter.yaml file for building our bundle. Not sure if that possibly has something to do with it? Might be relevant information though. Let me know what other info I may be able to provide (fairly new to porter, so  just let me know what I may be able to get you).
I appreciate your help!

**Reply by carolynvs** [reply]
I'm not sure yet what causes it but I know I have seen it occur transiently before. My guess is that there is a bit of a disconnect between what is published to docker hub, what's in the local docker cache and what porter has in it's own cache.

Thanks for the tip that you were using image digests in the bundle, I'll make sure to include that in my tests.

**Reply by samwise-commits** [reply]
No problem at all! I'm going to try adding in the digest tags again and see if it changes anything at all. I'll let you know what I find out.

**Reply by samwise-commits** [reply]
I was able to do a test last night with the porter.yaml file having the digests defined. I build a new bundle and then tried to archive it. I didn't get any errors but there was no output. I then ran it with `--debug` and the tgz file was populated!
Thought it odd, but left it at that. I tried rerunning this morning without debug and got the same empty tgz. Threw debug back in there and got an empty tgz this time. No idea what is happening, but there are no errors at all. Does the --debug option put a file somewhere that may have something in it?

**Reply by carolynvs** [reply]
Debug only controls what is printed to stdout, so this is really odd!

**Comment by carolynvs** [comment]
Here is a link to the PR where we are working on a fix https://github.com/getporter/porter/pull/1537
